apps/sms/test/marionette/drafts_test.js

```diff
@@ -145,51 +123,181 @@ marionette('Messages Drafts', function() {
         shouldHaveAttachment: true
       };
 
-      messagesStorage = [{
-        id: smsMessage.threadId,
-        body: smsMessage.body,
-        lastMessageType: smsMessage.type,
-        timestamp: smsMessage.timestamp,
-        messages: [smsMessage],
-        participants: [smsMessage.receiver]
-      }];
-
       messagesApp.launch();
-      storage.setMessagesStorage(messagesStorage);
+
+      storage.setMessagesStorage(
+        [conversation], ThreadGenerator.uniqueMessageId
+      );
+
+      inboxView = new InboxView(client);
     });
 
-    test('Draft is correctly saved', function() {
-      messagesApp.Inbox.navigateToComposer();
+    suite('Thread-less drafts', function() {
+      var newMessageView, draftConversationId;
 
-      createAndSaveDraft(draft);
+      setup(function() {
+        newMessageView = inboxView.createNewMessage();
+        createAndSaveDraft(newMessageView, draft);
 
-      assertDraft(draft);
+        // Let's wait for the draft node to appear.
+        client.waitFor(function() {
+          return inboxView.conversations.length === 2;
+        });
 
-      // Relaunch application and verify draft is persisted.
-      messagesApp.close();
-      messagesApp.launch();
-      storage.setMessagesStorage(messagesStorage);
+        draftConversationId = inboxView.conversations[0].id;
```

Comment:
I think you could waitForElement('.is-draft') (or something similar) and get the id from this element directly; it would be more robust than taking the first element ?

I don't exactly get why we wait for 2 threads btw.